File: survival-graphics/src/main/java/net/survival/graphics/Assets.java
package net.survival.graphics;

public final class Assets
{
    private static TextureAtlas mipmappedTextureAtlas;
    private static TextureAtlas textureAtlas;
    private static boolean initialized;

    private Assets() {}

    public static void setup(TextureAtlas mipmappedTextureAtlas, TextureAtlas textureAtlas) {
        if (initialized) {
            throw new IllegalStateException("Assets have already been initialized.");
        }

        Assets.mipmappedTextureAtlas = mipmappedTextureAtlas;
        Assets.textureAtlas = textureAtlas;
        initialized = true;
    }

    public static void tearDown() {
        if (!initialized) {
            throw new IllegalStateException("Assets have already been torn down.");
        }

        mipmappedTextureAtlas.close();
        textureAtlas.close();
        initialized = false;
    }

    public static TextureAtlas getMipmappedTextureAtlas() {
        if (mipmappedTextureAtlas == null) {
            throw new IllegalStateException("mipmappedTextureAtlas has not been initialized.");
        }

        return mipmappedTextureAtlas;
    }

    public static TextureAtlas getTextureAtlas() {
        if (textureAtlas == null) {
            throw new IllegalStateException("textureAtlas has not been initialized.");
        }

        return textureAtlas;
    }
}

File: survival-graphics/src/main/java/net/survival/graphics/Assets.kt
package net.survival.graphics

object Assets {
    private var mipmappedTextureAtlas: TextureAtlas? = null
    private var textureAtlas: TextureAtlas? = null
    private var initialized = false

    fun setup(mipmappedTextureAtlas: TextureAtlas?, textureAtlas: TextureAtlas?) {
        check(!initialized) { "Assets have already been initialized." }
        Assets.mipmappedTextureAtlas = mipmappedTextureAtlas
        Assets.textureAtlas = textureAtlas
        initialized = true
    }

    fun tearDown() {
        check(initialized) { "Assets have already been torn down." }
        mipmappedTextureAtlas!!.close()
        textureAtlas!!.close()
        initialized = false
    }

    fun getMipmappedTextureAtlas(): TextureAtlas {
        checkNotNull(mipmappedTextureAtlas) { "mipmappedTextureAtlas has not been initialized." }
        return mipmappedTextureAtlas!!
    }

    fun getTextureAtlas(): TextureAtlas {
        checkNotNull(textureAtlas) { "textureAtlas has not been initialized." }
        return textureAtlas!!
    }
}